lib/config_helpers.js

```diff
@@ -0,0 +1,160 @@
+"use strict";
+
+const assert = require('assert');
+const fs     = require('fs');
+
+const constants = require('./constants');
+
+const MASK = '*****';
+
+let _validateAuthConfiguration = ( opts ) => {
+  let authurl = opts.get('authurl');
+  if ( authurl ) {
+    if ( !authurl.startsWith('https') ) {
+      throw Error('Only HTTPS protocol should be used for auth server');
+    }
+
+    ['auth_cert', 'auth_key'].forEach( optname => {
+      assert(
+        opts.get(optname),
+        `'${optname}' is required when using an https Auth-server URL`
```

Comment:
docs/server.rst 

> If the listening server is expecting for npg_ranger to establish the connection using pre-arranged certificates, --auth_cert and --auth_key options should be used.

 suggests that client side certification is optional, but here we throw an error if cert and key aren't set